src/Service/AvailableCarService.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Service;
+
+use App\Entity\Booking;
+use App\Entity\Car;
+use App\Enum\EnumBookingStatus;
+use App\Service\PricingApi\PricingClientInterface;
+use Doctrine\ORM\EntityManagerInterface;
+
+class AvailableCarService
+{
+    public function __construct(
+        private readonly EntityManagerInterface $entityManager,
+        private readonly PricingClientInterface $pricingClient,
+    ) {
+    }
+
+    public function getAvailableCars(
+        int $stationId,
+        \DateTimeInterface $startDate,
+        \DateTimeInterface $endDate,
+    ): array {
+        $cars = $this->entityManager->getRepository(Car::class)->findBy([
+            'active' => true,
+        ]);
+
+        $availableCars = [];
+
+        foreach ($cars as $car) {
+            if ($car->getStation()?->getId() !== $stationId) {
+                continue;
+            }
+
+            $overlappingBookings = $this->entityManager
+                ->getRepository(Booking::class)
+                ->createQueryBuilder('b')
+                ->where('b.car = :car')
+                ->andWhere('b.startDate < :to')
+                ->andWhere('b.endDate > :from')
+                ->andWhere('b.status IN (:statuses)')
+                ->setParameters([
+                    'car' => $car,
+                    'from' => $startDate,
+                    'to' => $endDate,
+                    'statuses' => array_map(
+                        fn (EnumBookingStatus $status) => $status->value,
+                        EnumBookingStatus::blockingStatuses()
+                    ),
+                ])
+                ->getQuery()
+                ->getResult();
+
+            if (0 === \count($overlappingBookings)) {
+                $availableCars[] = $car;
+            }
+        }
+
+        return array_map(fn (Car $car) => [
+            'id' => $car->getId(),
+            'model' => $car->getModel(),
+            'station' => $car->getStation()?->getName(),
+            'price' => $this->pricingClient->calculatePrice(
+                $car->getId(),
+                $startDate,
+                $endDate,
+            ),
+        ], $availableCars);
+    }
```

Comment:
```suggestion
      return false;
    }
```
Returning different types (array|bool)